firebase-database/src/main/java/com/google/firebase/database/core/utilities/Utilities.java

```diff
@@ -73,6 +81,13 @@ public static ParsedUrl parseUrl(String url) throws DatabaseException {
       originalPathString = originalPathString.replace("+", " ");
       Validation.validateRootPathString(originalPathString);
 
+      // TODO: Should log this out
+      if (serviceSettings != null) {
+        repoInfo.secure = false;
+        repoInfo.host = serviceSettings.host + ":" + serviceSettings.port;
+        repoInfo.internalHost = repoInfo.host;
+      }
+
```

Comment:
As for comments for here, I think this code is fine. A cleaner way might be https://gist.github.com/schmidt-sebastian/e1ddba17d468ae72c4104fa84aae2d9a, but it's optional.

Reply:
You're right that is better, I implemented that diff.